internal/controller/groupqualityofservicelimits/groupqualityofservicelimits.go

```diff
@@ -198,11 +245,73 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalDelete{}, errors.New(errNotGroupQualityOfServiceLimits)
 	}
 
-	fmt.Printf("Deleting: %+v", cr)
+	cr.SetConditions(xpv1.Deleting())
+
+	user := cloudian.User{
+		GroupID: cr.Spec.ForProvider.GroupID,
+		UserID:  "*",
+	}
+	err := c.cloudianService.DeleteQOS(ctx, user, cr.Spec.ForProvider.Region)
+	// For consistency/futureproofing - API currently returns ok for non-existent groups
+	if err != nil && !errors.Is(err, cloudian.ErrNotFound) {
+		return managed.ExternalDelete{}, errors.Wrap(err, errGetCreds)
+	}
 
 	return managed.ExternalDelete{}, nil
 }
 
 func (c *external) Disconnect(ctx context.Context) error {
 	return nil
 }
+
+func toCloudianQos(qos v1alpha1.GroupQualityOfServiceLimitsParameters) (cloudian.QualityOfService, error) {
+	warning, err := toCloudianLimits(qos.Warning)
+	if err != nil {
+		return cloudian.QualityOfService{}, err
```

Comment:
```suggestion
		return nil, err
```

Better use a pointer here?

Reply:
Would cause a deref here, but will always be safe: `SetQOS(ctx, user, cr.Spec.ForProvider.Region, *qos)